tests/test_backend.py

```diff
@@ -31,174 +31,166 @@ def reverse_key(key):
+        "rediss://localhost:3333?db=2",
+    ],
+)
+def test_connection_strings(connection_string):
+    cf = pool.get_connection_factory(
+        path="django_redis.pool.ConnectionFactory", options={}
+    )
+    res = cf.make_connection_params(connection_string)
+    assert res["url"] == connection_string
+
+
+@pytest.fixture()
+def available_caches(settings):
+    caches_setting = copy.deepcopy(settings.CACHES)
+    caches_setting["default"]["KEY_PREFIX"] = "*"
+    settings.CACHES = caches_setting
+    cache = caches["default"]
+    other = caches["with_prefix"]
+    yield cache, other
+    cache.clear()
+    other.clear()
+
+
+class TestDjangoRedisCacheEscapePrefix:
+    def test_delete_pattern(self, available_caches):
+        cache, other = available_caches
+        cache.set("a", "1")
+        other.set("b", "2")
+        cache.delete_pattern("*")
+        assert cache.has_key("a") is False
+        assert other.get("b") == "2"
+
+    def test_iter_keys(self, available_caches):
+        cache, other = available_caches
+        if isinstance(cache.client, ShardClient):
+            pytest.skip("ShardClient doesn't support iter_keys")
 
+        cache.set("a", "1")
+        other.set("b", "2")
+        assert list(cache.iter_keys("*")) == ["a"]
 
-class DjangoRedisCacheTestCustomKeyFunction(unittest.TestCase):
-    def setUp(self):
-        caches_setting = copy.deepcopy(settings.CACHES)
-        caches_setting["default"]["KEY_FUNCTION"] = "test_backend.make_key"
-        caches_setting["default"]["REVERSE_KEY_FUNCTION"] = "test_backend.reverse_key"
-        cm = override_settings(CACHES=caches_setting)
-        cm.enable()
-        self.addCleanup(cm.disable)
+    def test_keys(self, available_caches):
+        cache, other = available_caches
+        cache.set("a", "1")
+        other.set("b", "2")
+        keys = cache.keys("*")
+        assert "a" in keys
+        assert "b" not in keys
 
-        self.cache = caches["default"]
 
-    def tearDown(self):
-        self.cache.clear()
+def test_custom_key_function(settings):
+    caches_setting = copy.deepcopy(settings.CACHES)
+    caches_setting["default"]["KEY_FUNCTION"] = "test_backend.make_key"
+    caches_setting["default"]["REVERSE_KEY_FUNCTION"] = "test_backend.reverse_key"
+    settings.CACHES = caches_setting
+    cache = caches["default"]
 
-    def test_custom_key_function(self):
-        if isinstance(self.cache.client, ShardClient):
-            self.skipTest("ShardClient doesn't support get_client")
+    if isinstance(cache.client, ShardClient):
+        pytest.skip("ShardClient doesn't support get_client")
 
-        for key in ["foo-aa", "foo-ab", "foo-bb", "foo-bc"]:
-            self.cache.set(key, "foo")
+    for key in ["foo-aa", "foo-ab", "foo-bb", "foo-bc"]:
+        cache.set(key, "foo")
 
-        res = self.cache.delete_pattern("*foo-a*")
-        self.assertTrue(bool(res))
+    res = cache.delete_pattern("*foo-a*")
+    assert bool(res)
```

Comment:
better to check if it is True
```suggestion
    assert res is True
```